master/buildbot/steps/source/p4.py

```diff
@@ -307,7 +309,11 @@ def _createClientSpec(self):
         def has_whitespace(*args):
             return any([re.search(r'\s', i) for i in args if i is not None])
 
-        if self.p4viewspec:
+        if self.stream:
+            q = '"' if has_whitespace(self.p4base, self.p4branch) else ''
+            client_spec += "Stream:\t%s%s/%s%s\n\n" % (q, self.p4base, self.p4branch, q)
```

Comment:
why not quoting it unconditionally?

I see that the rest of the code does that as well, but sounds weird anyway

Reply:
Agreed. Also, can we use python3's `.format()`?

Reply:
My code is similar to the code that can be found a few lines below it (and was known to be working).

https://github.com/buildbot/buildbot/blob/20b7713d18501e4b41235d87e7a5f4cb1a531728/master/buildbot/steps/source/p4.py#L327

https://github.com/buildbot/buildbot/blob/20b7713d18501e4b41235d87e7a5f4cb1a531728/master/buildbot/steps/source/p4.py#L349

If I update my code, should I update the rest of the code too ?

Reply:
> If I update my code, should I update the rest of the code too ?

yes please. you probably will need to also update the tests

Reply:
String formatting was fixed in another PR.

I did a few test with perforce and it appears that you can't have whitespace in a Stream path (example: in **//depot/stream** both **depot** and **stream** can't have whitespace). I added a check and removed the quote handling.

I don't think I should change the way the quoting is implemented in the rest of code for now. Maybe in a future PR.